standard/unsafe-code.md

```diff
@@ -277,6 +277,7 @@ In an unsafe context, the set of available implicit conversions ([§10.2](conver
 
 - From any *pointer_type* to the type `void*`.
 - From the `null` literal ([§6.4.5.7](lexical-structure.md#6457-the-null-literal)) to any *pointer_type*.
+- From any *pointer_type* to the type `System.Span<T>`, where `T` is the referent type of *pointer_type*.
```

Comment:
This objection looks reasonable - and if we can just change it to an implicit conversion from *stackalloc_expression*, I *think* that should be simple. But I don't know the origin of this change, so that could be missing something.

Reply:
Agree with @KalleOlaviNiemitalo, the conversion from a pointer must be wrong.

An implicit conversion from a *stackalloc_expression* to `Span<T>` is not required as that is the type of such an expression in safe code, from §12.8.21:

> The result of a *stackalloc_expression* is an instance of type `Span<T>`, where `T` is the *unmanaged_type*…

In unsafe code it can be a pointer or a `Span<T>`, from §23.9:

> In an unsafe context if a *stackalloc_expression* (§12.8.21) occurs as the initializing expression of a *local_variable_declaration* (§13.6.2), where the *local_variable_type* is either a pointer type (§23.3) or inferred (`var`), then the result of the *stackalloc_expression* is a pointer of type `T *` to be beginning of the allocated block, where `T` is the *unmanaged_type* of the *stackalloc_expression*.
>
> In all other respects the semantics of *local_variable_declarations* (§13.6.2) and *stackalloc_expressions* (§12.8.21) in unsafe contexts follow those defined for safe contexts.

The added conversion wouldn’t make sense even in unsafe code, unless the resulting span had infinite length (and couldn’t be passed to safe code!)

Just delete this added conversion.